### PR TITLE
refactor!: credentialSchema does not work for CTypes

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#2747

We finally have a proper draft spec for JSON-schema descriptions of VCs - but this is not particularly compatible with our CTypes, and the latest CType version we released is even less likely to work with this.

So I wanted to explore what it would look like if we used this field for something else - namely for describing what a KiltCredentialV1 type VC has to look like. CType verification would then have to be moved elsewhere; I think it's best here to just do it as part of credential verification.

__Done so far:__
- [x] Update VC exporter to output a credentialSchema where the type is `JsonSchema2023` and the id references the our credential schema for KiltCredentialV1 type credentials.
- [x] Require this credentialSchema on verification.
- [x] Perform CType validation on proof verification instead, with automatic fetching of the respective CType definition enabled by default (can be deactivated)
- [x] Add option to verification methods to specify the expected CType(s), which in combination with disabling the CType loader enforces that credentials follow these CType(s).
- [x] Define a new field on which the CType of a Kilt VC is listed and easily accessible (e.g. credential `type` field?).  

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: packages/vc-export/src/KiltAttestationProofV1.ts [hunk 1]
```diff
@@ -616,7 +620,9 @@ export function initializeProof(
     proof,
     [
       rootHash,
-      CType.idToHash(credentialSchema.id as ICType['$id']),
+      CType.idToHash(
+        type.find((str) => str.startsWith('kilt:ctype:')) as ICType['$id']
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/vc-export/src/KiltAttestationProofV1.ts:624]: but a credential using this type of proof can only ever have one CType as far as I can see. If there were multiple, how would we decide which one is "the" type of the credential that we expect, for example, in the attestation record?

File: packages/vc-export/src/CredentialSchema.ts [hunk 2]
```diff
@@ -11,39 +11,95 @@ import type { ICType } from '@kiltprotocol/types'
...
+    credentialSubject,
+    type,
+  }: Pick<KiltCredentialV1, 'credentialSubject' | 'type'>,
+  {
+    cTypes = [],
+    loadCTypes = cachingCTypeLoader,
+  }: { cTypes?: ICType[]; loadCTypes?: false | CTypeLoader } = {}
+): Promise<void> {
+  // get CType id referenced in credential
+  const credentialsCTypeId = type.find((str) =>
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/vc-export/src/CredentialSchema.ts:66]: same as above. In fact, the JSON-Schema that checks the KiltCredentialV1 structure purposefully disallows more than one CType, or any additional entries in the type field.

**Review by ntn-x2 — APPROVED** [review]
Your comments make sense.

**Comment by rflechtner** [comment]
@ntn-x2 I did have a concern that `CredentialSchema.validateSubject` may be outdated as an accessor to a function verifying a credential's subject against it's (C)type due to the proposed change. This is because this check now has nothing to do with the VC's `credentialSchema` property. What's your opinion on this? Should we move this to `KiltCredentialV1`? Or do you  have another suggestion for where this functionality should live?

**Comment by ntn-x2** [comment]
> What's your opinion on this? Should we move this to `KiltCredentialV1`?

It would make sense, right? This is pretty much tied to what we do in this version of the spec.

**Pull request merged**